kld/src/ldk/peer_manager.rs

```diff
@@ -1,78 +1,127 @@
 use std::{net::SocketAddr, sync::Arc, time::Duration};
 
 use crate::api::NetAddress;
+use crate::bitcoind::{BitcoindClient, BitcoindUtxoLookup};
 use crate::database::{peer::Peer, LdkDatabase};
-use crate::settings::Settings;
-use anyhow::{anyhow, bail, Context, Result};
+use crate::logger::KldLogger;
+use anyhow::{anyhow, Context, Result};
+use async_trait::async_trait;
 use bitcoin::secp256k1::PublicKey;
-use log::{error, info};
+use ldk_lsp_client::LiquidityManager;
+use lightning::sign::KeysManager;
+use lightning::{
+    ln::{channelmanager::SimpleArcChannelManager, peer_handler},
+    onion_message::SimpleArcOnionMessenger,
+    routing::gossip,
+};
+use lightning_net_tokio::SocketDescriptor;
+use log::{error, info, warn};
 use tokio::task::JoinHandle;
 
-use super::{ChannelManager, LdkPeerManager};
+use super::{ChainMonitor, ChannelManager, KldRouter};
 
-pub struct PeerManager {
-    ldk_peer_manager: Arc<LdkPeerManager>,
-    channel_manager: Arc<ChannelManager>,
-    database: Arc<LdkDatabase>,
-    settings: Arc<Settings>,
-    addresses: Vec<NetAddress>,
-}
+pub(crate) type PeerManager = peer_handler::PeerManager<
+    SocketDescriptor,
+    Arc<SimpleArcChannelManager<ChainMonitor, BitcoindClient, BitcoindClient, KldLogger>>,
+    Arc<
+        gossip::P2PGossipSync<
+            Arc<gossip::NetworkGraph<Arc<KldLogger>>>,
+            Arc<BitcoindUtxoLookup>,
+            Arc<KldLogger>,
+        >,
+    >,
+    Arc<SimpleArcOnionMessenger<KldLogger>>,
+    Arc<KldLogger>,
+    Arc<
+        LiquidityManager<
+            Arc<KeysManager>,
+            Arc<ChainMonitor>,
+            Arc<BitcoindClient>,
+            Arc<BitcoindClient>,
+            Arc<KldRouter>,
+            Arc<KeysManager>,
+            Arc<KldLogger>,
+            Arc<KeysManager>,
+        >,
+    >,
+    Arc<KeysManager>,
+>;
+
+#[async_trait]
+pub trait KuutamoPeerManger {
+    async fn listen(&self, port: u16) -> Result<()>;
+    async fn connect_peer(
+        &self,
+        database: Arc<LdkDatabase>,
+        public_key: PublicKey,
+        peer_addr: NetAddress,
+    ) -> Result<()>;
 
-impl PeerManager {
-    pub fn new(
-        ldk_peer_manager: Arc<LdkPeerManager>,
+    fn keep_channel_peers_connected(
+        &self,
+        database: Arc<LdkDatabase>,
         channel_manager: Arc<ChannelManager>,
+    );
+
+    // Regularly broadcast our node_announcement. This is only required (or possible) if we have
+    // some public channels, and is only useful if we have public listen address(es) to announce.
+    // In a production environment, this should occur only after the announcement of new channels
+    // to avoid churn in the global network graph.
+    fn regularly_broadcast_node_announcement(&self, node_alias: String, addresses: Vec<NetAddress>);
```

Comment:
As the comment says, this will spam the network. Leave it as it is. Only broadcast after a channel opening or config change (restart).

Reply:
The way we call `regularly_broadcast_node_announcemen` does not change. The function just moves into the trait.

Reply:
The implementation of this function has a 60 second loop, this is wrong, it should not be called. But you are not calling this function now anyway so it can be deleted.

Reply:
function still needs to be deleted